ext4/tar2ext4/tar2ext4.go

```diff
@@ -85,23 +93,32 @@ func ConvertTarToExt4(r io.Reader, w io.ReadWriteSeeker, options ...Option) erro
 			return err
 		}
 
-		if err = fs.MakeParents(hdr.Name); err != nil {
-			return errors.Wrapf(err, "failed to ensure parent directories for %s", hdr.Name)
+		name := hdr.Name
+		linkName := hdr.Linkname
+		if p.convertBackslash {
+			// compactext assumes all paths are `/` separated
+			// [filepath.ToSlash]  will be a nop on Linux, but replace '\' with '/' on Windows.
+			name = filepath.ToSlash(name)
+			linkName = filepath.ToSlash(linkName)
```

Comment:
```suggestion
			name = strings.ReplaceAll(name, `\`, "/")
			linkName = strings.ReplaceAll(linkName, `\`, "/")
```

We should do this unconditionally, rather than depending on what OS we are running on. We want to be able to support processing a Windows tar on Linux and vice-versa.